controllers/controllers/clusters/vsphere.go

```diff
@@ -135,5 +137,49 @@ func (v *VSphereClusterReconciler) Reconcile(ctx context.Context, cluster *anywh
 		return reconciler.Result{}, err
 	}
 
+	workerNodeGroupMachineSpecs := make(map[string]anywherev1.VSphereMachineConfigSpec, len(machineConfigMap))
+	for _, wnConfig := range cluster.Spec.WorkerNodeGroupConfigurations {
+		workerNodeGroupMachineSpecs[wnConfig.MachineGroupRef.Name] = machineConfigMap[wnConfig.MachineGroupRef.Name].Spec
+	}
+
+	cp := machineConfigMap[specWithBundles.Spec.ControlPlaneConfiguration.MachineGroupRef.Name]
+	var etcd *anywherev1.VSphereMachineConfigSpec
+	if specWithBundles.Spec.ExternalEtcdConfiguration != nil {
+		etcdSpec := machineConfigMap[specWithBundles.Spec.ExternalEtcdConfiguration.MachineGroupRef.Name]
+		etcd = &etcdSpec.Spec
+	}
+
+	templateBuilder := vsphere.NewVsphereTemplateBuilder(&dataCenterConfig.Spec, &cp.Spec, etcd, workerNodeGroupMachineSpecs, time.Now, true)
+	clusterName := cluster.ObjectMeta.Name
+
+	cpOpt := func(values map[string]interface{}) {
+		values["controlPlaneTemplateName"] = templateBuilder.CPMachineTemplateName(clusterName)
+		controlPlaneUser := machineConfigMap[cluster.Spec.ControlPlaneConfiguration.MachineGroupRef.Name].Spec.Users[0]
+		values["vsphereControlPlaneSshAuthorizedKey"] = controlPlaneUser.SshAuthorizedKeys[0]
+		etcdUser := machineConfigMap[cluster.Spec.ExternalEtcdConfiguration.MachineGroupRef.Name].Spec.Users[0]
+		values["vsphereEtcdSshAuthorizedKey"] = etcdUser.SshAuthorizedKeys[0]
+		values["etcdTemplateName"] = templateBuilder.EtcdMachineTemplateName(clusterName)
+	}
+	v.Log.Info("cluster", "name", cluster.Name)
+	controlPlaneSpec, err := templateBuilder.GenerateCAPISpecControlPlane(specWithBundles, cpOpt)
+	if err != nil {
+		return reconciler.Result{}, err
+	}
+
+	workersSpec, err := templateBuilder.GenerateCAPISpecWorkers(specWithBundles, nil)
+	if err != nil {
+		return reconciler.Result{}, err
+	}
+
+	if err := reconciler.ReconcileYaml(ctx, v.Client, controlPlaneSpec); err != nil {
+		return reconciler.Result{Result: &ctrl.Result{
+			Requeue:      true,
+			RequeueAfter: time.Minute,
```

Comment:
can we maybe use `const` for requeue time?

Reply:
added a constant, but this is a temporary code and will probably change